src/app/apps/rup/mapa-camas/sidebar/desocupar-cama/desocupar-cama.component.ts

```diff
@@ -95,18 +95,25 @@ export class CamaDesocuparComponent implements OnInit, OnDestroy {
         this.camaOcupada$ = combineLatest([
             this.mapaCamasService.selectedCama,
             this.mapaCamasService.snapshot$,
+            this.historial$
         ]).pipe(
-            map(([selectedCama, snapshots]) => {
-                const cama = snapshots.find(snap => snap.idCama === selectedCama.idCama);
-                return (cama.estado !== 'ocupada' || cama.idInternacion !== selectedCama.idInternacion) && !cama.sala;
+            map(([selectedCama, snapshots, historial]) => {
+                // En caso de no haber seleccionado ninguna cama pasamos al segundo if donde vamos a verificar si
+                // la cama esta ocupada en el ultimo movimiento del historial de movimientos.
+                const historialPaciente = historial[historial?.length - 1];
+                if (selectedCama.idCama) {
+                    const cama = snapshots.find(snap => snap.idCama === selectedCama.idCama);
+                    return (cama.estado !== 'ocupada' || cama.idInternacion !== selectedCama.idInternacion) && !cama.sala;
+                } else if (historialPaciente?.idCama) {
+                    const cama = snapshots.find(snap => snap.idCama === historialPaciente.idCama);
+                    return (cama.estado !== 'ocupada' || cama.idInternacion !== historialPaciente.idInternacion) && !cama.sala;
+                }
```

Comment:
```suggestion
                const historialPaciente = historial[historial?.length - 1];
                const camaPaciente = selectedCama || historialPaciente;
                const cama = snapshots.find(snap => snap.idCama === camaPaciente.idCama);
                return (cama.estado !== 'ocupada' || cama.idInternacion !== camaPaciente.idInternacion) && !cama.sala;
```
Cuando dependiendo de una condicion el codigo que sigue es el mismo pero con diferente variable, conviene "empaquetar" este valor para no repetir codigo mas adelante (Como muestra la sugerencia).

Volviendo al codigo original:
1. Se da a entender que las condiciones de las lineas 104 y 107 podrian no cumplirse. Es asi? que retornaria en todo caso?
2. En la linea 103, el '?' da a entender que el historial puede venir varcio. Si no es asi sería mejor sacarlo para no esconder algun posible bug.

Reply:
Corregido!! se agregaron las nuevas sugerencias y se modifico el nombre del observable para mayor comprensión.